src/trade/liquidation.ts

```diff
@@ -29,13 +35,68 @@ export const getLiquidationPrice = (
     initialAccFees,
     context
   );
-  const liqPriceDistance =
+  const liqThresholdP = getLiqPnlThresholdP(
+    context.liquidationParams,
+    trade.leverage,
+    context.contractsVersion
+  );
+
+  let liqPriceDistance =
     (trade.openPrice *
-      (trade.collateralAmount * 0.9 - (borrowingFee + closingFee))) /
+      (trade.collateralAmount * liqThresholdP - (borrowingFee + closingFee))) /
     trade.collateralAmount /
     trade.leverage;
 
+  if (
+    context?.contractsVersion === ContractsVersion.V9_2 &&
+    context?.liquidationParams?.maxLiqSpreadP !== undefined &&
+    context.liquidationParams.maxLiqSpreadP > 0
+  ) {
+    const closingSpreadP = getSpreadP(
+      context.pairSpreadP,
+      true,
+      context.liquidationParams
+    );
+
+    liqPriceDistance -= trade.openPrice * closingSpreadP;
+  }
+
   return trade.long
     ? Math.max(trade.openPrice - liqPriceDistance, 0)
     : Math.max(trade.openPrice + liqPriceDistance, 0);
 };
+
+export const getLiqPnlThresholdP = (
+  liquidationParams: LiquidationParams | undefined,
+  leverage: number | undefined,
+  contractsVersion: ContractsVersion | undefined
+): number => {
+  if (
+    liquidationParams === undefined ||
+    leverage === undefined ||
+    contractsVersion === ContractsVersion.BEFORE_V9_2 ||
```

Comment:
This is inaccurate, if a trade opened before v9.2 uses partial add, its liquidation params will be refreshed.